### PR TITLE
PE-883: rename-file CLI

### PR DESCRIPTION
**Pull request by matibat**

### CODE REVIEW

File: src/commands/rename_file.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			wallet: wallet,
+			feeMultiple: parameters.getOptionalBoostSetting(),
+			dryRun
+		});
+
+		const result = await (async function () {
+			if (await parameters.getIsPrivate()) {
+				const driveId = await ardrive.getDriveIdForFileId(fileId);
+				const driveKey = await parameters.getDriveKey({ driveId });
+				const file = await ardrive.getPrivateFile({
```
Comment (fedellen) [inline src/commands/rename_file.ts:36]: As I commented in the other PR, consider handling this in the ArDrive layer for an easier to use API

**Comment by fedellen** [comment]
Reminder to add a README component here

File: src/commands/index.ts [hunk 2]
```diff
@@ -23,6 +23,7 @@ import './list_drive';
 import './list_folder';
 import './move_file';
 import './move_folder';
+import './rename_file';
```
Comment (arielmelendez) [inline src/commands/index.ts:26]: Thank you for preserving alphabetical order here :)

File: src/commands/rename_file.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import { CLIAction } from '../CLICommand/action';
+import { EID, Wallet } from 'ardrive-core-js';
+
+new CLICommand({
+	name: 'rename-file',
+	parameters: [FileIdParameter, FileNameParameter, BoostParameter, DryRunParameter, ...DrivePrivacyParameters],
+	action: new CLIAction(async function action(options) {
+		const parameters = new ParametersHelper(options);
+
+		const dryRun = !!parameters.getParameterValue(DryRunParameter);
```
Comment (arielmelendez) [inline src/commands/rename_file.ts:20]: We do this double negation dance in so many places that I think it's time for a `isDryRun` function on ParametersHelper.

File: src/commands/rename_file.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+		const fileId = parameters.getRequiredParameterValue(FileIdParameter, EID);
+		const newName = parameters.getRequiredParameterValue(FileNameParameter);
+
+		const wallet: Wallet = await parameters.getRequiredWallet();
+		const owner = await wallet.getAddress();
```
Comment (arielmelendez) [inline src/commands/rename_file.ts:25]: You won't need to get this if you follow my advice in the ArFSDao and ArDrive PRs.

File: src/commands/rename_file.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+				const file = await ardrive.getPrivateFile({
+					fileId,
+					driveKey,
+					owner
+				});
+
+				return ardrive.renamePrivateFile({
+					file,
+					newName,
+					metaDataRewardSettings: { feeMultiple: parameters.getOptionalBoostSetting() },
```
Comment (arielmelendez) [inline src/commands/rename_file.ts:45]: Interesting that we'd need to do this even though we set the boost setting to the ArDrive class already. Is it really necessary to have in the API?

**Review by arielmelendez — CHANGES_REQUESTED** [review]
See my comments.

File: src/commands/rename_file.ts (hunk 4 above)
Comment (matibat) [reply src/commands/rename_file.ts]: Code cleanup: 806cdbe

File: src/commands/rename_file.ts (hunk 5 above)
Comment (matibat) [reply src/commands/rename_file.ts]: Code cleanup: 806cdbe

File: src/commands/rename_file.ts (hunk 3 above)
Comment (matibat) [reply src/commands/rename_file.ts]: Implements isDryRun and adds a couple of tests: 8306eac

File: package.json [hunk 6]
```diff
@@ -8,7 +8,7 @@
 	},
 	"types": "./lib/index.d.ts",
 	"dependencies": {
-		"ardrive-core-js": "1.6.0",
+		"ardrive-core-js": "../ardrive-core-js_PE-879",
```
Comment (matibat) [inline package.json:11]:
```suggestion
		"ardrive-core-js": "../ardrive-core-js_PE-881",
```

File: package.json [hunk 7]
```diff
@@ -8,7 +8,7 @@
 	},
 	"types": "./lib/index.d.ts",
 	"dependencies": {
-		"ardrive-core-js": "1.6.0",
+		"ardrive-core-js": "../ardrive-core-js_PE-881",
```
Comment (arielmelendez) [inline package.json:11]: Reminder to update this when ready

**Review by arielmelendez — DISMISSED** [review]
CLEAN!

**Review by arielmelendez — APPROVED** [review]

**Review by fedellen — DISMISSED** [review]
🙌

**Review by fedellen — APPROVED** [review]

**Review by fedellen — APPROVED** [review]

**Pull request merged**